parlai/core/torch_classifier_agent.py

```diff
@@ -168,6 +170,127 @@ def value(self) -> float:
             return numer / denom
 
 
+class AUCMetrics(Metric):
+    """
+    Class that calculates the area under the roc curve from list of labels and its true
```

Comment:
nit: docstrings should have a single sentence intro, then a more detailed description

```
Computes Area Under ROC Curve (AUC) metrics.

Does so by keeping track of positives and negatives. Note the introduction of
`max_bucket_dec_places`, which is used for ...
```